### PR TITLE
LP-2275 Fix linting issues in badging app

### PR DESCRIPTION
**Pull request by zeshanarifios**
### Description

[LP-2275](https://philanthropyu.atlassian.net/browse/LP-2275)

[philu-edx-theme PR](https://github.com/philanthropy-u/philu-edx-theme/pull/1420)

- [x] Fix all the linting issues
- [x] Resolve cyclic import issue by changing helpers structure
- [x] Fix already broken unit tests in badging app
- [x] Run isort in badging app

### CODE REVIEW

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

File: openedx/features/badging/helpers/badges.py [hunk 1]
```diff
@@ -1,25 +1,23 @@
+"""
+Helper methods
```
Comment (zahid-arbisoft) [inline openedx/features/badging/helpers/badges.py:2]: Lets improve docstring by saying `All helper methods specifically related to badges`

or If you have anything better in mind, please add.

File: openedx/features/badging/tabs.py [hunk 2]
```diff
@@ -1,4 +1,9 @@
+"""
+This module is essentially a broker to `xmodule/tabs.py`, It was originally introduced
+to display My Badges in Course Experience side menu for the badging feature
```
Comment (zahid-arbisoft) [inline openedx/features/badging/tabs.py:3]: `... to display "My Badges" in course experience side...`

Or

`... to display my badges in course experience side...`

File: openedx/features/badging/templatetags/badge_progress.py [hunk 3]
```diff
@@ -1,3 +1,6 @@
+"""
+Template tags related to badge progress and congratulations modal
```
Comment (zahid-arbisoft) [inline openedx/features/badging/templatetags/badge_progress.py:2]: `...congratulations modal popup` or `...congratulations modal dialog`

File: openedx/features/badging/views.py [hunk 4]
```diff
@@ -1,30 +1,39 @@
...
 from .models import UserBadge
-from .constants import BADGES_KEY, COURSES_KEY, COURSE_ID_KEY, COMMUNITY_URL_KEY, COURSE_NAME_KEY
 
 
 @require_GET
 @login_required
 def trophycase(request):
+    """
+    This displays list of all badges in enrolled courses
+    """
```
Comment (zahid-arbisoft) [inline openedx/features/badging/views.py:31]: Other than tests, we add a complete docstring. Please update other occurrences, if any

 ```
   """
    This view displays list of all badges in enrolled courses

    Args:
        request (HttpRequest): Django request object

    Returns:
        HttpResponse: Template with context
    """
```

**Review by zahid-arbisoft — CHANGES_REQUESTED** [review]
Please run isort on the following files

```
/edx/app/edxapp/edx-platform/openedx/features/badging/tests/factories.py
/edx/app/edxapp/edx-platform/openedx/features/badging/tests/test_assign_missing_badges.py
/edx/app/edxapp/edx-platform/openedx/features/badging/tests/test_helpers.py
/edx/app/edxapp/edx-platform/openedx/features/badging/tests/test_views.py
```

File: openedx/features/badging/handlers.py [hunk 5]
```diff
@@ -12,7 +15,7 @@
 
 
 @receiver(post_save, sender=Badge)
-def sync_badge_info_with_nodebb(sender, instance, update_fields, **kwargs):
+def sync_badge_info_with_nodebb(sender, instance, update_fields, **kwargs):  # pylint: disable=unused-argument
```
Comment (harris-irfan) [inline openedx/features/badging/handlers.py:18]: Instead of disabling this pylint warning in 4 different places within the file, we can do it once at the start, before the import statements, since there are total of 4 functions in this file, and this warning is to be disabled for all 4 functions. WDYT?

File: openedx/features/badging/models.py [hunk 6]
```diff
@@ -38,6 +41,9 @@ def get_badges_json(self, badge_type):
 
 
 class Badge(models.Model):
+    """
+    This stores the badge information
```
Comment (harris-irfan) [inline openedx/features/badging/models.py:45]: Maybe we can rephrase it to:

"Model to store badge information"

File: openedx/features/badging/tabs.py [hunk 7]
```diff
@@ -1,4 +1,9 @@
+"""
+This module is essentially a broker to `xmodule/tabs.py`, It was originally introduced
```
Comment (harris-irfan) [inline openedx/features/badging/tabs.py:2]: This module is essentially a broker to `xmodule/tabs.py`**.** It was originally introduced

File: openedx/features/badging/tests/test_helpers.py [hunk 8]
```diff
@@ -94,14 +100,14 @@ def test_get_course_badges_with_empty_badge_queryset(self, mock_badge_objects_al
...
-    @mock.patch('openedx.features.badging.helpers.get_course_by_id')
+    @mock.patch('openedx.features.badging.helpers.badges.add_badge_earned_date')
+    @mock.patch('openedx.features.badging.helpers.badges.filter_earned_badge_by_joined_team')
+    @mock.patch('openedx.features.badging.helpers.badges.is_teams_feature_enabled')
+    @mock.patch('openedx.features.badging.helpers.badges.get_course_by_id')
     def test_get_course_badges_successfully(self, mock_get_course_by_id,
                                         mock_is_teams_feature_enabled,
                                         mock_filter_earned_badge_by_joined_team,
-                                        mock_add_badge_earned_date):
+                                        mock_add_badge_earned_date):  # pylint: disable=unused-argument
```
Comment (harris-irfan) [inline openedx/features/badging/tests/test_helpers.py:110]: Please update this according to [edx recommendations](https://edx.readthedocs.io/projects/edx-developer-guide/en/latest/style_guides/python-guidelines.html#breaking-long-lines).

File: openedx/features/badging/tests/test_helpers.py [hunk 9]
```diff
@@ -144,14 +160,14 @@ def test_get_course_badges_successfully(self, mock_get_course_by_id,
...
+    @mock.patch('openedx.features.badging.helpers.badges.filter_earned_badge_by_joined_team')
+    @mock.patch('openedx.features.badging.helpers.badges.is_teams_feature_enabled')
+    @mock.patch('openedx.features.badging.helpers.badges.get_course_by_id')
     def test_get_course_badges_user_not_joined_any_course_team(self, mock_get_course_by_id,
-                                                           mock_is_teams_feature_enabled,
-                                                           mock_filter_earned_badge_by_joined_team,
-                                                           mock_add_badge_earned_date):
+                                                       mock_is_teams_feature_enabled,
+                                                       mock_filter_earned_badge_by_joined_team,
+                                                       mock_add_badge_earned_date):  # pylint: disable=unused-argument
```
Comment (harris-irfan) [inline openedx/features/badging/tests/test_helpers.py:170]: Please update according to [edx recommendations](https://edx.readthedocs.io/projects/edx-developer-guide/en/latest/style_guides/python-guidelines.html#breaking-long-lines).

**Review by harris-irfan — CHANGES_REQUESTED** [review]

File: openedx/features/badging/handlers.py (hunk 5 above)
Comment (zeshanarifios) [reply openedx/features/badging/handlers.py:18]: I think it is fine to disable for these places only instead of disabling for whole files.

File: openedx/features/badging/tabs.py [hunk 10]
```diff
@@ -1,6 +1,6 @@
 """
-This module is essentially a broker to `xmodule/tabs.py`, It was originally introduced
-to display My Badges in Course Experience side menu for the badging feature
+This module is essentially a broker to `xmodule/tabs.py`. It was originally introduced
+to display "My Badges" in Course Experience side menu for the badging feature
```
Comment (zahid-arbisoft) [inline openedx/features/badging/tabs.py:3]: Let's change Course Experience to `course experience`

**Review by zahid-arbisoft — APPROVED** [review]

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

File: openedx/features/badging/tests/test_helpers.py [hunk 11]
```diff
@@ -105,9 +106,8 @@ def test_get_course_badges_with_empty_badge_queryset(self, mock_badge_objects_al
     @mock.patch('openedx.features.badging.helpers.badges.is_teams_feature_enabled')
     @mock.patch('openedx.features.badging.helpers.badges.get_course_by_id')
     def test_get_course_badges_successfully(self, mock_get_course_by_id,
-                                        mock_is_teams_feature_enabled,
-                                        mock_filter_earned_badge_by_joined_team,
-                                        mock_add_badge_earned_date):  # pylint: disable=unused-argument
+            mock_is_teams_feature_enabled, mock_filter_earned_badge_by_joined_team,
+            mock_add_badge_earned_date):  # pylint: disable=unused-argument
```
Comment (harris-irfan) [inline openedx/features/badging/tests/test_helpers.py:110]: It's still not according to [edx recommendations](https://edx.readthedocs.io/projects/edx-developer-guide/en/latest/style_guides/python-guidelines.html#breaking-long-lines). It should be like:

`    def test_get_course_badges_successfully(
        self,
        mock_get_course_by_id,
        mock_is_teams_feature_enabled,
        mock_filter_earned_badge_by_joined_team,
        mock_add_badge_earned_date  # pylint: disable=unused-argument`
    ):

Please see if statement to disable pylint warning is correctly placed on the above code snippet.

Please update accordingly in lines 168-169 as well.

**Review by harris-irfan — CHANGES_REQUESTED** [review]

File: openedx/features/badging/tests/test_helpers.py (hunk 11 above)
Comment (zeshanarifios) [reply openedx/features/badging/tests/test_helpers.py:110]: The link you are referring to is for calling the function not to define a function. Please go through [pep8](https://www.python.org/dev/peps/pep-0008/) 
```
# Add 4 spaces (an extra level of indentation) to distinguish arguments from the rest.
```

Comment (harris-irfan) [reply openedx/features/badging/tests/test_helpers.py:110]: Ok, then lets please follow the edx recommendation. Right now the arguments are over intended (there should be 1 extra level of indentation before arguments according to what you shared. Right now there are about 13 extra levels of indentation).

Please update on both places.

Comment (zeshanarifios) [reply openedx/features/badging/tests/test_helpers.py:110]: Please see the examples in pep8 and compare it with my code. I hope it will be helpful to understand. 

here is example code
```
# Add 4 spaces (an extra level of indentation) to distinguish arguments from the rest.
def long_function_name(
        var_one, var_two, var_three,
        var_four):
    print(var_one)
```

There is an extra level of indentation from method body. That's what I am following.

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Review by harris-irfan — APPROVED** [review]

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python
